content/en/docs/pipelines/tutorials/benchmark-examples.md

```diff
@@ -0,0 +1,184 @@
+'storage.cloud.google.com' will incure a redirect and doesn't work well with
+Kubeflow Pipelines.
+>
+> **num_runs** specifies how many runs will be created in the benchmark script,
+and is a direct indicator of the simulated workload.
+> - **NOTE**: please be aware that running the benchmark script on a cloud
+service will incur possible charges due to the consumption of cloud resources.
+Therefore, please stay within your cloud resource budget if any.
+>
+> **run_status_polling_interval_sec** sets the time interval between two
+adjacent pollings of the run status. When a run reaches the success status, the
+run duration is recorded. Only the durations of succeeded runs are recorded.
+
+After the values of the above parameters are properly set, you can run the
+benchmark script in the notebook. The following snapshot shows the
+results of running the benchmark script [run_service_api.ipynb](https://github.com/jingzhang36/pipelines/blob/different_tools/tools/benchmarks/run_service_api.ipynb)
+with **num_runs** being 50 on a Kubernetes cluster with two node pools, while
+each node pool has three nodes of machine type 'n1-standard-8'.
+
+<img src="/docs/images/benchmark-snapshot-1.png"
+alt="Benchmark Sample Output Plots"
+class="mt-3 mb-3 border border-info rounded">
+
+## Interpretation of Results
+
+In the above example output, there are two types of plots. One is the
+distribution plot, for latency and duration measurement; the other is the count
+plot, for couting succeeded and failed runs. The reading of those plots is in
+general straightfoward.
+
+In a count plot, the x-axis repsresents the possible run status: success or fail;
+and the y-axis shows how many runs fall into certain status respectively.
+
+In a distribution plot, both histogram plot and rug plot are shown. In addtion,
+it is also possible to show KDE (Kernel Density Estimate) plot. If a KDE plot is
+desirable, please use 'kde=True' in the distplot() method.
+
+## Limitations And Future Work
+
+When the benchmark script is tuned to generate a moderate workload, e.g., 50
+runs in the above example, the latency and run duration measurements can be
+made properly. However, it is also interesting to see how the Kubeflow Pipelines
+deployment will behave or break under some extremely heavy workloads, or in
+other words, to probe the Kubeflow Pipelines deployment. The example benchmark
+script can be used for that purpose as well. In that case, the measurement plots
+of the benchmark script are no longer the expected output, but instead, the
+potential errors and error logs are expected to provide information on the
+performance and scalibity of the Kubeflow Pipelines deployment. With them, bugs
+and pain points can be discovered and then fixed. Moreover, when probing the
+Kubeflow Pipelines deployment with extreme workloads, it will be really helpful
+to add internal monitoring to the server code to track server performance. E.g.,
```

Comment:
Suggested revision:

For example, in the future, it would be desirable to use [Prometheus](https://prometheus.io/) to track and visualize metrics for Kubeflow Pipelines servers.